db/migrate/20170530102506_add_deleted_to_containers_tables.rb

```diff
@@ -0,0 +1,25 @@
+class AddDeletedToContainersTables < ActiveRecord::Migration[5.0]
+  def change
+    add_column :container_definitions, :deleted, :boolean, :default => false, :null => false
+    add_column :container_groups, :deleted, :boolean, :default => false, :null => false
+    add_column :container_images, :deleted, :boolean, :default => false, :null => false
+    add_column :container_projects, :deleted, :boolean, :default => false, :null => false
+    add_column :containers, :deleted, :boolean, :default => false, :null => false
+
+    add_index :container_definitions, :deleted,
+              :name  => "index_container_definitions_on_deleted_false",
+              :where => "NOT deleted"
```

Comment:
For a boolean it's recommended. Since PG would not pick a normal index on a boolean, because the cost is too high.